### PR TITLE
Update README

### PR DESCRIPTION
**Pull request by vkostyukov**
I tried to describe the errors API in Finagle Serial. I wasn't thinking about the implementation, I was thinking about user experience. The idea is to replace `ClientError` with `CodecError` and throw/serialize it every time we have troubles with codec. This should be very clear for users. Also, it would be great to get rid of `ServerApplicationError` and replace within an actual exception throw by user. This will make Finagle Serial looks like other Finagle codecs.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -61,34 +61,32 @@ Scodec Support
...
- * `com.twitter.finagle.mux.ServerError`
- * `com.twitter.finagle.mux.ServerApplicationError`.
+* `io.github.finagle.serial.CodecError`
+* `com.twitter.finagle.mux.ServerError`
 
-Both of the exceptions indicate server side errors. `ServerError` indicates a failure in the
-transport layer (i.e., failed serialization or deserialization of the request or response),
-while `ServerApplicationError` indicates a failure in user defined code (i.e., service implementation).
+A `CodecError` indicates that codec fails to serialize or deserialize either
+request or response. The exception may be thrown by either client or server.
```
Comment (travisbrown) [inline README.md:71]: Nitpick: "returned" might be better than "thrown", since in any case this will always be in a `Try`.

**Comment by vkostyukov** [comment]
I'm going abandon this for now.

**Comment by travisbrown** [comment]
Sounds good—I'm aiming to get a draft in about the current approach soon. The basic idea for `CodecError` is the same (although it's now called `EncodingError`). Each implementation can decide which application errors it wants to encode, and any others have their messages wrapped in an `UnencodedError`.

**Comment by vkostyukov** [comment]
I like new approach very much. The only thing that confused me at the first time is exception names. At the first glance, it's not really clean what that `UnencodedError` is for domain exceptions but not for exceptions from `decode` phase. What do you think about changing the names a bit, to have:
- `CodecError` - literally represents an error happened in the codec implementation (encode/decode failures)
- `ApplicationError` - represents an error in the user application (code); uses to wrap domain exceptions if no suitable codec was found

I think these names will make the things clearer. Probably, we don't need to change the names just because I didn't get them in the first time.

**Pull request closed without merging**